### DISCUSSION TITLE
Webpack 5 cache "filesystem" breaks app with default cacheFolder.

### DISCUSSION THREAD

**Discussion by olegKusov** [webpack/webpack#18649] [Q&A]
# Bug report

**What is the current behavior?**

I used persistent cache (cache filesystem) with DndProvider component (https://react-dnd.github.io/react-dnd/) a lot. But now got error in production build for my testing environment when using DndProvider on page while running browser test. 

```
Objects are not valid as a React child (found: object with keys {$$typeof, type, key, ref, props}). If you meant to render a collection of children, use an array instead.
```

Funny thing that i tried it on my cloud Linux dev machine and there is no problem. Now it happens only on my local machine, so the problem with cache i guess.

if cache: false, then error do not exists. I do not get the reason why it worked for a while and now even on old commits it do not work. I fully cleaned all node_modules files (.cache folder is inside node_modules) and pnpm caches, i removed fully my project and pulled it from git from scratch - did not helped. 

**Only thing that helped is setting cache_directory:**
```
{
                  type: 'filesystem',
                  cacheDirectory: path.resolve(__dirname, '.temp_cache'),
}
 ```

**If the current behavior is a bug, please provide the steps to reproduce.**

1. add to webpack config:
 ```
cache:  {
            type: 'filesystem',
        }
``` 
        

**What is the expected behavior?**
I expect no error to appear when we run application in testing environment (browser tests).

**Other relevant information:**
webpack version:  "5.91.0",
Node.js version: v20.13.1
Operating System: OS: macos 14.1.1

**Comment by alexander-akait** [comment]
Please create reproducible test repo, your error is not related to cache

> Funny thing that i tried it on my cloud Linux dev machine and there is no problem. Now it happens only on my local machine, so the problem with cache i guess.

Sounds like you have something with your local env, maybe you have different version of library
